scripts/console_container.gd

```diff
@@ -117,35 +120,33 @@ func _show_autocomplete(text: String) -> void:
 		_tooltip_label.set_text("\n".join(autocomplete))
 		_tooltip_label.show()
 
-
 func _on_input_text_changed(text: String) -> void:
-	_autocomplete_index = 0
-	_show_autocomplete(text)
-
+	_show_autocomplete(_autocomplete_prompt if _autocomplete_prompt.length() > 0 else text)
 
 func _cycle_autocomplete(direction: int) -> void:
-	var autocomplete: PackedStringArray = _console.autocomplete_list(_console_input.get_text())
+	var autocomplete: PackedStringArray = _console.autocomplete_list(_autocomplete_prompt)
 	_autocomplete_index = wrapi(_autocomplete_index + direction, 0, autocomplete.size())
-
-	_show_autocomplete(_console_input.get_text())
+	_show_autocomplete(_autocomplete_prompt)
+	set_input_text(_console.autocomplete_command(_autocomplete_prompt, _autocomplete_index))
 
 func _on_input_gui_event(event: InputEvent) -> void:
+	if event.is_action_type() && !event.is_action(&"ui_text_indent"):
+		# start below zero so the first tab press "selects" the autocomplete
```

Comment:
GDScript style guide [says](https://docs.godotengine.org/en/latest/tutorials/scripting/gdscript/gdscript_styleguide.html#boolean-operators) to use plain english.
```suggestion
	if event.is_action_type() and not event.is_action(&"ui_text_indent"):
		# Start below zero so the first tab press "selects" the autocomplete.
```